src/site/content/en/blog/codelab-sign-in-form-best-practices/index.md

```diff
@@ -0,0 +1,371 @@
+</div>
+
+Click **View App** to check out your freshly styled sign-in form. Then
+click **View Source** to return to `style.css`.
+
+That's quite a lot of code! The main things to be aware of are the changes to sizes:
+
+* `padding` and `margin` are added to inputs.
+* `font-size` is different for mobile and desktop.
+
+The `:invalid` selector is used to indicate when an input has an invalid value.
+
+The CSS layout is mobile-first:
+
+* The default CSS is for viewports less than `450px` wide.
+* The media query section sets overrides for viewports that are at least `450px` wide.
+
+It's very important at this point to test your code on real devices on desktop and mobile:
+
+* Is label and input text readable, especially for people with low vision?
+* Are the inputs and **Log in** button large enough to use as touch targets for thumbs?
+
+## Step 3: Add input attributes to enable built-in browser features
+
+Enable the browser to store and autofill input values, and provide access to 
+built-in password management features.
+
+Add attributes to your form HTML so it looks like this:
+
+```html/3,4,7,8,10
+<form action="#" method="post">
+  <h1>Sign in</h1>
+  <section>        
+    <label for="email">Email</label>
+    <input id="email" name="email" type="email" placeholder=" " autocomplete="email" required autofocus>
+  </section>
+  <section>        
+    <label for="current-password">Password</label>
+    <input id="password" name="password" type="password" autocomplete="new-password" required>
+  </section>
+  <button id="sign-in">Sign in</button>
+</form>
+```
+
+Click the label that says **Email**. Notice how focus moves to the email input.
+This is because the label is associated with the input via the `for="email"` attribute.
+Screenreaders also announce the label text when the label or the label's associated
+input gets focus.
+
+Try focusing the email input on a mobile device. Notice how the keyboard is optimized
+for typing an email address. For example, the `@` and `.` characters might be shown on
+the primary keyboard, and the operating system might show stored emails above the keyboard.
+All of this happens because the `type="email"` attribute is applied to an `<input>` element.
+
+<figure class="w-figure">
+  <img class="w-screenshot w-screenshot--filled" 
+       src="email-keyboard.png" 
+       alt="The default email keyboard on iOS.">
+</figure>
+
+Try typing some text into the password input. The text is hidden by default because the
+`type="password"` attribute has been applied to the element.
+
+* `name="email"` and `name="current-password"` help browsers store named values 
+which can later be used for `autocomplete`.
+
+Try focusing the email input on a desktop device and start typing. If you've 
+stored any email addresses in your browser, you'll probably see a popup that 
+allows you to select from those stored emails. This happens because the 
+`autocomplete="email"` attribute was applied to the email input.
```

Comment:
```suggestion
Try focusing the email input on a desktop device and start typing.
You can find the URL of your app by clicking **Fullscreen**
![The Fullscreen icon](/images/glitch/fullscreen.svg). 
If you've stored any email addresses in your browser, you'll probably see a popup that 
allows you to select from those stored emails. This happens because the 
`autocomplete="email"` attribute was applied to the email input.
```